docs/DeveloperGuide.md

```diff
@@ -657,37 +657,235 @@ testers are expected to do more *exploratory* testing.
+
+    4. Other incorrect edit task commands to try: `edit exam`, `edit task m/NON_EXISTENT_MODULE`, `...`<br>
+     Expected: Similar to previous.
+
+### Deleting a task
+
+1. Deleting a task from a module (and from all its students) while all modules are being shown.
+
+    1. Prerequisites: List all modules using the `home` command. At least 1 `Module` is in the TAB and that `Module` has at lest 1 `Task`. The task will still be deleted if there are no `Student`s in the `Module`, but the change will only be visible after a `Student` has been added.
+
+    2. Test case: `delete task m/CS2103 ti/T1`<br>
+       Expected: The `Task` with the task id `T1` is deleted from the `Module` `CS2103`.
+
+    3. Test case: `delete task m/NON_EXISTENT_MODULE`<br>
+       Expected: No `Task` is deleted. Error details shown in the status message.
+
+    4. Other incorrect delete task commands to try: `delete assignment`, `delete task m/CS2103 i/NON_EXISTENT_TASK_ID`, `...`<br>
+       Expected: Similar to previous.
+
+### Marking a task as done
+
+1. Marking a task from a student in a particular module as done (`Completed`).
+
+   1. Prerequisites: List all modules using the `home` command. The `Task` being marked done must belong to a `Student` in a `Module` in TAB and the `Task` must be `Incomplete`.
+   
+   2. Test case: `mark done m/CS2103 i/A1234567A ti/T1`<br>
+      Expected: The `Task` with the task id `T1` is marked as `Completed` for `Student` with student id `A1234567A` of `Module` `CS2103`.
+   
+   3. Test case: `mark done m/CS2103 i/A1234567A ti/T1` (again, after the task is already marked as `Completed`)<br>
+      Expected: No `Task` is marked as `Completed`. Error details shown in the status message.
+   
+   4. Other incorrect mark done commands to try: `mark finished`, `mark done m/NON_EXISTENT_MODULE i/NON_EXISTENT_STUDENT_ID ti/NON_EXISTENT_TASK_ID`, `...`<br>
       Expected: Similar to previous.
+
+### Marking a task as undone
+
+1. Marking a task from a student in a particular module as undone (`Incomplete`).
+
+    1. Prerequisites: List all modules using the `home` command. The `Task` being marked undone must belong to a `Student` in a `Module` in TAB and the `Task` must be `Completed`.
+
+    2. Test case: `mark undone m/CS2103 i/A1234567A ti/T1`<br>
+       Expected: The `Task` with the task id `T1` is marked as `Incomplete` for `Student` with student id `A1234567A` of `Module` `CS2103`.
+
+    3. Test case: `mark undone m/CS2103 i/A1234567A ti/T1` (again, after the task is already marked as `Incomplete`)<br>
+       Expected: No `Task` is marked as `Completed`. Error details shown in the status message.
    
-### Saving data
+    4. Other incorrect mark undone commands to try: `mark unfinished`, `mark undone m/NON_EXISTENT_MODULE i/NON_EXISTENT_STUDENT_ID ti/NON_EXISTENT_TASK_ID`, `...`<br>
+       Expected: Similar to previous.
 
-1. Dealing with missing/corrupted data files
+### Finding a student
```

Comment:
should this one go with all the student related testings?

Reply:
which line do you mean?

Reply:
I think it's there cos it's before the home command, or maybe we can do module, task, then students? then it can flow nicely hehe

Reply:
I've moved the saving data segment up

Reply:
Oh i see, the ordering of the segments

Reply:
would need to keep find and home together, so i can shift both of those between the student and task segments?

Reply:
I think it's better to do Module -> Student -> Task, so I'll put the find and home between the student and task, would that be better?

Reply:
The current ordering is the same as the UG

Reply:
yes I think that would be better

Reply:
ok will shift those now